In [8]:
import pandas as pd
from scipy import stats

# Q1.

In [3]:
cutlet = pd.read_csv('Cutlets.csv')
cutlet

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

## 2 sample t test


###  H0: There is no signifincant difference between two units
### H1: There is signifincant difference between two units

In [6]:
cutlet['Unit A'].mean()

7.01909142857143

In [7]:
cutlet['Unit B'].mean()

6.964297142857142

In [13]:
statis,pvalue = stats.ttest_ind(a = cutlet['Unit A'], b= cutlet['Unit B'])
print('t-statistic:',statis)
print('pvalue:',pvalue)

t-statistic: 0.7228688704678063
pvalue: 0.4722394724599501


#### we reject the null hypothesis as pvalue (0.472) > 0.05. There is signifincant difference between two units

# Q2.

In [14]:
lab = pd.read_csv('LabTAT.csv')
lab

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

## 4 sample t test (Anova test)

### H0: There is no significant difference between the avg TAT of reports
### H1: There is  significant difference with atleast one of avg TAT of reports

In [20]:
tstats,pValue=stats.f_oneway(lab.iloc[:,0],lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])

In [21]:
print('t-statistic:',tstats)
print('pvalue:',pValue)

t-statistic: 118.70421654401437
pvalue: 2.1156708949992414e-57


#### we do not reject the null hypothesis as pvalue <  0.05. There is no significant difference between the avg TAT of reports

# Q3.

In [22]:
buyer = pd.read_csv('BuyerRatio.csv')
buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

## Chi square Test
### H0: P1 = P2 = P3 = P4
### H1: There is  significant difference with atleast one of buyer rations

In [64]:
buyer_table = buyer.iloc[:,1:]
buyer_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [73]:
buyer_table.dtypes

East     int64
West     int64
North    int64
South    int64
dtype: object

In [66]:
buyer_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [67]:
chi2_score,pval,dof,expected_table = stats.chi2_contingency(buyer_table)

In [68]:
print('Chi-squared value : ',round(chi2_score,5))
print('P-val             : ',round(pval,5))
print('Degree of Freedom : ',dof)
print('Expected Table    :\n',expected_table)

Chi-squared value :  1.59595
P-val             :  0.66031
Degree of Freedom :  3
Expected Table    :
 [[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


In [72]:
if pval <= 0.05:
    print('1 = P2 = P3 = P4')
else:
    print('There is  significant difference with atleast one of buyer rations')

There is  significant difference with atleast one of buyer rations


# Q4.

In [74]:
order = pd.read_csv('Costomer+OrderForm.csv')
order

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [80]:
order.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [81]:
order.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [83]:
order.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [85]:
order.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [87]:
obv = np.array([[271,267,269,280],[29,33,31,20]])
obv

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [89]:
chi2_score,pval,dof,expected_table = stats.chi2_contingency(obv)

In [90]:
print('Chi-squared value : ',round(chi2_score,5))
print('P-val             : ',round(pval,5))
print('Degree of Freedom : ',dof)
print('Expected Table    :\n',expected_table)

Chi-squared value :  3.85896
P-val             :  0.2771
Degree of Freedom :  3
Expected Table    :
 [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [91]:
if pval <= 0.05:
    print('1 = P2 = P3 = P4')
else:
    print('The defective percentage varies by centre')

The defective percentage varies by centre
